lib/diego/client.rb

```diff
@@ -9,19 +11,28 @@ class Client
     def initialize(url:, ca_cert_file:, client_cert_file:, client_key_file:,
       connect_timeout:, send_timeout:, receive_timeout:)
       ENV['PB_IGNORE_DEPRECATIONS'] ||= 'true'
+      @config = VCAP::CloudController::Config.config
       @client = build_client(
-        url,
         ca_cert_file,
         client_cert_file,
         client_key_file,
         connect_timeout,
         send_timeout,
         receive_timeout)
+      @locket_client = Locket::Client.new(
```

Comment:
During my "experiments" (*) I got the impression that the DNS resolution is done only once for a locket client. The BOSH DNS query type is different compared to `bbs.service.cf.internal`, i.e. it "should" return healthy instances only, but I think that it might be better to move the locket client creation into a dedicated method and initialize a fresh client for every `fetch` call.

(*) What I tested was adding a general network delay to one locket VM and I could see that the client was either using the fast or the slow locket service throughout its lifetime.

Reply:
If I understand correctly, you want to ensure that Locket::Client re-resolves DNS on each of the three retries, and your concern is that, without instantiating a new 'stub' each time, those retries would just keep trying the same unavailable server?

Reply:
Ok this is sorted